src/arrayMethodReduce.js

```diff
@@ -4,8 +4,20 @@
  * Implement method Reduce
  */
 function applyCustomReduce() {
-  [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+  [].__proto__.reduce2 = function(callback, startValue) {
+    let previousValue = startValue;
+    let startIndex = 0;
+
+    if (arguments.length < 2) {
+      previousValue = this[0];
+      startIndex = 1;
```

Comment:
you can use ternary to set the value of the start index and you wont need this if block